### PR TITLE
Add support for JSON patches in folder spec files

### PR DESCRIPTION
**Pull request by mrodm**
## What does this PR do?

This PR adds support to define [JSON patches](https://jsonpatch.com/) at folder specs (`spec.yml`) files too.
It also ensure that JSON patches are defined in `versions` key keeping the newer versions on top.

It also removes usages of `errors.Wrapf` in files under `code/go/internal/specschema` module.

## Why is it important?

This is important to allow us to remove/add/replace files or folders depending on the spec version used by the package.

## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/main/test/packages) that prove my change is effective.
    - Add tests in `specschema` and `validator` modules
- [x] I have added an entry in [`spec/changelog.yml`](https://github.com/elastic/package-spec/blob/main/spec/changelog.yml).

## Related issues

- Relates #514

### CODE REVIEW

**Comment by mrodm** [comment]
test integrations

**Review by jsoriano — APPROVED** [review]
:+1:

**Comment by mrodm** [comment]
test integrations

File: code/go/internal/specschema/folder_schema_spec.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	var resolved folderItemSpec
+	err = json.Unmarshal(spec, &resolved)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal resolved spec: %w", err)
+	}
+	return &resolved, nil
+}
+
+func (f *folderSchemaSpec) patchForVersion(target semver.Version) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (mrodm) [reply code/go/internal/specschema/folder_schema_spec.go:44]: I did some refactor to extract that logic and create a new module `specpatch`.

**Comment by mrodm** [comment]
Failures in the integrations PR https://github.com/elastic/integrations/pull/6587 are not related to any issue with package validation.

**Review by jsoriano — APPROVED** [review]

**Pull request merged**